File: codegen/results/v002/qwen3.6_35b_think/20260418_171211/typescript/002_word_frequency/solution.ts
import * as fs from 'fs';

const content = fs.readFileSync('input/text.txt', 'utf-8');
const lowerContent = content.toLowerCase();
const words = lowerContent.match(/[a-z]+/g) || [];

const freq = new Map<string, number>();
for (const word of words) {
  freq.set(word, (freq.get(word) || 0) + 1);
}

const result = Array.from(freq.entries())
  .sort((a, b) => {
    if (b[1] !== a[1]) return b[1] - a[1];
    return a[0].localeCompare(b[0]);
  });

for (const [word, count] of result) {
  console.log(`${word}: ${count}`);
}

File: codegen/results/v002/qwen3.6_35b_think/20260418_171211/python/002_word_frequency/solution.py
import re
from collections import Counter

def main():
    # Read the file
    with open('input/text.txt', 'r') as f:
        text = f.read()

    # Convert to lowercase
    text = text.lower()

    # Strip punctuation (keep only letters)
    # Replace any character that is not a lowercase letter with a space
    clean_text = re.sub(r'[^a-z]', ' ', text)

    # Split into words (handles multiple spaces resulting from punctuation removal)
    words = clean_text.split()

    # Count frequencies
    counts = Counter(words)

    # Sort by count descending, then alphabetically ascending for ties
    # key is (-count, word)
    sorted_counts = sorted(counts.items(), key=lambda item: (-item[1], item[0]))

    # Output
    for word, count in sorted_counts:
        print(f"{word}: {count}")

if __name__ == "__main__":
    main()